src/DataProtection/custom/Cmdlets/Platform/Vault/Set-AzDataProtectionMSIPermission.ps1

```diff
@@ -0,0 +1,205 @@
+    [CmdletBinding(PositionalBinding=$false)]
+    [Microsoft.Azure.PowerShell.Cmdlets.DataProtection.Description('Initializes Backup instance Request object for configuring backup')]
+
+    param(
+        [Parameter(Mandatory, HelpMessage='Backup instance request object which will be used to configure backup')]
+        [Microsoft.Azure.PowerShell.Cmdlets.DataProtection.Models.Api20220401.IBackupInstanceResource]
+        ${BackupInstance},
+        
+        [Parameter(Mandatory=$false, HelpMessage='ID of the keyvault')]
+        [ValidatePattern("/subscriptions/([A-z0-9\-]+)/resourceGroups/(?<rg>.+)/(?<id>.+)")]
+        [System.String]
+        ${KeyvaultId},
+
+        [Parameter(Mandatory, HelpMessage='Resource group of the backup vault')]
+        [System.String]
+        ${VaultResourceGroup},
+        
+        [Parameter(Mandatory, HelpMessage='Name of the backup vault')]
+        [System.String]
+        ${VaultName},
+
+        [Parameter(Mandatory, HelpMessage='Scope at which the permissions need to be granted')]
+        [System.String]
+        [ValidateSet("Resource","ResourceGroup","Subscription")]
+        ${PermissionsScope}
+        
+    )
+
+    process {
+          CheckResourcesModuleDependency
+          CheckPostgreSqlModuleDependency         
+          
+          $DatasourceId = $BackupInstance.Property.DataSourceInfo.ResourceId
+          $DatasourceType = $BackupInstance.Property.DataSourceInfo.ResourceType
+
+          if($DatasourceType -eq "Microsoft.Storage/storageAccounts"){$DatasourceType = "AzureBlob"}
+          elseif($DatasourceType -eq "Microsoft.Compute/disks"){$DatasourceType = "AzureDisk"}
+          elseif($DatasourceType -eq "Microsoft.DBforPostgreSQL/servers/databases"){$DatasourceType = "AzureDatabaseForPostgreSQL"}
+                 
+          $manifest = LoadManifest -DatasourceType $DatasourceType.ToString()
+
+          $ResourceArray = $DataSourceId.Split("/")
+          $ResourceRG = "/subscriptions/" + $ResourceArray[2] + "/resourceGroups/" + $ResourceArray[4]
+          $SubscriptionName = "/subscriptions/" + $ResourceArray[2]
+          $vault = Get-AzDataProtectionBackupVault -VaultName $VaultName -ResourceGroupName $VaultResourceGroup -SubscriptionId $ResourceArray[2]
+          
+          $AllRoles =  Get-AzRoleAssignment -ObjectId $vault.IdentityPrincipalId
+
+          $MissingRolesInitially = $false
+
+          foreach($Permission in $manifest.keyVaultPermissions)
+          {              
+              Write-Host "We will attempt to automatically grant the foloowing access: `n
+                            1.Backup vault’s identity access on the Postgres server and the key vault `n
+                            2.'Allow All Azure services' under network connectivity in the Key vault `n"
+
+              $Choice = Read-Host "Press n/N to exit the grant permissions or press any other key to continue"
```

Comment:
Is this the standard way of prompting input from user in PS? Also if the user wants to bypass these prompts is there any way?

Reply:
Ideally there should be a parameter in the command which user can use to bypass all the prompts